crates/vm/levm/src/opcode_handlers/system.rs

```diff
@@ -254,4 +255,116 @@ impl VM {
+        // Returns: VMError RevertOpcode if executed correctly.
+        // Notes:
+        //      The reversion of changes is made in the generic_call().
+        //      Changes are not "reverted" if it is the first callframe, they are just not commited.
+
+        let offset = current_call_frame
+            .stack
+            .pop()?
+            .try_into()
+            .map_err(|_| VMError::VeryLargeNumber)?;
+
+        let size = current_call_frame
+            .stack
+            .pop()?
+            .try_into()
+            .map_err(|_| VMError::VeryLargeNumber)?;
+
+        let gas_cost = current_call_frame.memory.expansion_cost(offset + size)?;
+
+        self.increase_consumed_gas(current_call_frame, gas_cost)?;
+
+        current_call_frame.returndata = current_call_frame.memory.load_range(offset, size).into();
+
+        Err(VMError::RevertOpcode)
+    }
+
+    /// ### INVALID operation
+    /// Reverts consuming all gas, no return data.
+    pub fn op_invalid(&mut self) -> Result<OpcodeSuccess, VMError> {
+        Err(VMError::InvalidOpcode)
+    }
+
+    // SELFDESTRUCT operation
+    pub fn op_selfdestruct(
+        &mut self,
+        current_call_frame: &mut CallFrame,
+    ) -> Result<OpcodeSuccess, VMError> {
+        // Sends all ether in the account to the target address
+        // Steps:
+        // 1. Pop the target address from the stack
+        // 2. Get current account and: Store the balance in a variable, set it's balance to 0
+        // 3. Get the target account, checking if it is empty and if it is cold. Update gas cost accordingly.
+        // 4. Add the balance of the current account to the target account
+        // 5. Register account to be destroyed in accrued substate.
+
+        // Notes:
+        //      If context is Static, return error.
+        //      If executed in the same transaction a contract was created, the current account is registered to be destroyed
+        if current_call_frame.is_static {
+            return Err(VMError::OpcodeNotAllowedInStaticContext);
+        }
+
+        // Gas costs variables
+        let static_gas_cost = gas_cost::SELFDESTRUCT_STATIC;
+        let dynamic_gas_cost = gas_cost::SELFDESTRUCT_DYNAMIC;
+        let _cold_gas_cost = gas_cost::COLD_ADDRESS_ACCESS_COST;
+        let mut gas_cost = static_gas_cost; // This will be updated later
+
+        // 1. Pop the target address from the stack
+        let target_address = Address::from_low_u64_be(current_call_frame.stack.pop()?.low_u64());
+
+        // 2. Get current account and: Store the balance in a variable, set it's balance to 0
+        let current_account_balance = self
+            .db
+            .accounts
+            .get(&current_call_frame.to)
+            .unwrap()
+            .balance;
+        self.db
+            .accounts
+            .get_mut(&current_call_frame.to)
+            .unwrap()
+            .balance = U256::zero();
```

Comment:
In the new PR that implements Database changes the `db` works differently and if an account "doesn't exist" then it will return an empty account so there will be no need to handle this kind of errors. Maybe after merging that one to main I can make changes to this PR so that it fits the new database implementation.